sips/sip-122.md

```diff
@@ -142,16 +152,87 @@ In the event a synth has triggered a circuit breaker which pauses exchanging, th
+
+**New Naming Convention of ERC-20 Contracts**
+- Name of collateral contract and collateral state revised to include the token symbol
+
 ### Test Cases
 
-Included with implementation.
-[NEED TO INCLUDE: loan moved from under-collateralized to over-collateralized state within the same block, activates the interaction delay]
+#### Collapsible Shorts
+
+##### Context Liquidatable Debt Formula:
+- \\(c\\) = sUSD  Collateral of short
+- \\(d\\) = Value of shorters' debt in sUSD 
+- \\(t\\) = Minimum Collateralization Ratio
+- \\(l_a\\) = Amount of liquidatable debt in sUSD
+- \\(p_r\\) = Liquidation Penalty %
+- \\(f_r\\) = Fee Pool Reward Rate
+- \\(k_r\\) = Keeper Reward Rate
+
+\\[
+l_a = \frac{t * d - c}{t - (1 + p_r)}
+\\]
+
+##### Tests:
+- Given that `liquidateFromCollateral` is triggered 
+  - And that the loan has a collateralization ratio below 100%
+      - ✅ Then it succeeds and the following take place:
+	      - the debt of the borrower \\(d\\) is settled
+          - \\(p_{r}  k_{r} c)\\) from the loan collateral is sent to the `msg.sender`
+          - \\(c (1- p_{r} k_{r})\\) being the remainder of the loan collateral is sent to the fee pool
+          - loan is closed by liquidation
+        Effectively the loss from an under-water loan is distributed on all minters in proportion to their debt (socialized loss distribution)
+  - And that the loan has a collateralization ratio above 100% but strictly below 100% + \\(p_r\\)
+      - ✅ Then it succeeds and the following take place:
+	      - the loan balance \\(d\\) is settled
+          - \\(p_{r}  k_{r} c\\) from the loan collateral is sent to the `msg.sender`
+          - \\(c (1- p_{r} k_{r} )\\) being the remainder of the loan collateral is sent to the fee pool
+          - loan is closed by liquidation
+        Effectively the reward distributed to the fee pool is below \\(p_{r} c f_{r} \\) due to not enough collateral to satify this requirement but no socialized loss is borne by minters
+  - And that the loan has a collateralization ratio above or equal to 100% + \\(p_r\\) and strictly below \\(t\\)
+      - ✅ Then it succeeds and the following take place:
+	      - \\(l_{a}\\) of the borrowers' debt is settled
+          - \\( p_{r} l_{a} k_{r} \\) from the loan collateral is sent to the `msg.sender`
+          - \\( l_{a} (1+ p_{r} f_{r}) \\) from the loan collateral  is sent to the fee pool
+  - And that the loan has a collateralization ratio above or equal to \\(t\\)
+     - ❌ Then the transaction reverts due to the loan being above the minimum collateralization ratio
+        
+#### Interaction Delay Circumvention:
+- Given that a loan is below or equal to the min collateralization ratio \\(t\\) + 1% and is currently subject to the interaction delay timer    
+  - And that one of the below interactions is attempted:
+    - deposit collateral
+    - repayment of debt
```

Comment:
Allowing repayment of debt to circumvent the 1 hour interaction delay could create a loophole around the 1 hr protection that the short position / loan has to be open for at least 1 hour.

Reply:
yeah i had some time to think about it the 1% and i'm not to keen on it... the issue is that it opens up a 120-121% corridor where bad actors can start doing repay-drawdown-repay-drawdown to front run oracles so i'm going to remove the 1%. I was hoping to allow borrowers that they are able to close/repay right after the loan was liquidated (since liquidation to 120% triggers the 1 hour timer) but seems there is no easy way around it except coding in that if it is a liquidation event than reset the timer so that there is no lock up... However if it's too complicated and prone to a work-around attack, I think it's best we just reset the timer on liquidation like any interaction and probably the guy that got liquidated will need to wait it out, hoping that either his c-ratio will remain above 120% (no more liquidation) or his c-ratio will go down just slightly enough that liquidation doesn't make financial sense... Although if it's a big loan it's an issue

Reply:
@kaleb-keny  

I would suggest it would be much simpler and safer to not have the +1% buffer. 

You would permit the borrower / trader to repay their position to fix their c-ratio if it was at risk of being liquidated when its below 120%, still circumventing the 1 hour interaction delay if that is the case. 

This is because if someone wanted to open a short position right at 120% c-ratio backed, then they could close it bypassing the 1 hour delay at 121% ratio for example.

Reply:
alright i'll remove the 1% i agree it is un-necessary...
My concern was that they will wait until after they get liquidated to take action 😅  and people aren't as good at triggering as liquidation bots, so if they fall again below 120% they'll get hit again... But I think it's really an edge case and it opens up for more problems... Hopefully we can revisit this interaction buffer once we have fee recl in place... as then we don't need to worry about un-intended consequences

Reply:
alright it's done